api/src/test/java/org/openmrs/module/attachments/obs/TestHelper.java

```diff
@@ -294,4 +300,60 @@ public String getTestComplexObsFilePath() {
 		}
 		return obsList;
 	}
+	
+	/**
+	 * Boilerplate method to construct and save a complex obs with random data.
+	 * 
+	 * @param patient the patient
+	 * @param concept the associated concept
+	 * @return obs
+	 */
+	public Obs saveComplexObs(Patient patient, Concept concept) throws IOException {
+		Obs obs = new Obs();
+		byte[] randomData = new byte[20];
+		obs.setConcept(concept);
+		obs.setObsDatetime(new Date());
+		obs.setPerson(patient);
+		
+		new Random().nextBytes(randomData);
+		
+		String filename = RandomStringUtils.randomAlphabetic(7) + ".ext";
+		MockMultipartFile multipartRandomFile = new MockMultipartFile(FilenameUtils.getBaseName(filename), filename,
+		        "application/octet-stream", randomData);
+		obs.setComplexData(
+		    complexDataHelper.build(ValueComplex.INSTRUCTIONS_DEFAULT, multipartRandomFile.getOriginalFilename(),
+		        multipartRandomFile.getBytes(), multipartRandomFile.getContentType()).asComplexData());
+		
+		return context.getObsService().saveObs(obs, null);
+	}
+	
+	/**
+	 * Factory method that constructs and persists a ComplexConcept object
+	 * 
+	 * @return complexConcept
+	 */
+	public Concept createComplexConcept(String uuid, String name, String handler, String description) {
+		ConceptService conceptService = Context.getConceptService();
+		ConceptComplex conceptComplex = new ConceptComplex();
+		conceptComplex.setUuid(uuid);
+		conceptComplex.setHandler(handler);
+		ConceptName conceptName = new ConceptName(name, Locale.ENGLISH);
+		conceptComplex.setFullySpecifiedName(conceptName);
+		conceptComplex.setPreferredName(conceptName);
+		conceptComplex.setConceptClass(conceptService.getConceptClassByName("Question"));
+		conceptComplex.setDatatype(conceptService.getConceptDatatypeByUuid(ConceptDatatype.COMPLEX_UUID));
+		conceptComplex.addDescription(new ConceptDescription(description, Locale.ENGLISH));
+		
+		return conceptService.saveConcept(conceptComplex);
+	}
```

Comment:
Isn't there any such concept in the standard test dataset, or one of Core's datasets?

Reply:
Not to my knowledge. The standard dataset has limited complex concepts.